docs/DeveloperGuide.md

```diff
@@ -935,11 +941,70 @@ tutorial group as XLSX files. The resulting XLSX file will either contain attend
 each student across all tutorial lessons in the tutorial group. These features were designed with the user's needs in
 mind after surveying all the CS2101 tutors and interviewing a CS2101 tutor.
 
-### **G.2. Conclusion**
+### **G.2. Challenges**
+
+Throughout the development of **Serenity**, significant hurdles were faced in development.
+We detail some of the challenges faced, and steps we took to solve them.
+
+**Challenge 1: Design stage**
+
+The first challenge we faced involved the planning stage, where we had to have various
+different entities (tutorial groups, lessons, students and their records) come together.
+Compared to **AddressBook**, which had a single `Person` object in the model, we had to
+revamp it to accommodate multiple entities such as `Group`, `Lesson`, `Student` and `StudentInfo`, while
+ensuring we adhere to principles such as Separation of concern 
+and avoiding deep nesting.
+
+**Our solution to the challenge:**
+
+A key part towards solving this is laying down design requirements at the start
+of the project, to minimize the risk of refactoring and overhauling at 
+the end of our project. Requirements were clearly set up before any development began.
+Weekly team meetings were held to allocate tasks and plan for the week ahead.
+Within each week, we continued to update each other daily on our individual progress,
+to stay aware of the latest changes done in the project.
+We placed heavy emphasis on communication and put in a collaborative effort in ensuring
+that the project's challenges and tasks are clearly identified.
+
+**Challenge 2: Implementation**
+
+**Serenity** allows users to view the different aspects of their lessons, 
+such as viewing attendance records, participation records, lesson details and group details. 
+This posed a challenge in implementing an intuitive
+<span ><a href="#appendix-e-glossary" style="color:purple"><i>Graphical User Interface (GUI)</i></a></span> that also updated automatically
```

Comment:
You can just type "GUI" instead of "Graphical User Interface" as the full form of the word is shown in the Glossary